addons/im_livechat/controllers/main.py

```diff
@@ -150,14 +165,13 @@ def get_session(self, channel_id, anonymous_name, previous_operator_id=None, cha
                 chatbot_script._post_welcome_steps(channel)
             with replace_exceptions(UserError, by=NotFound()):
                 # sudo: mail.guest - creating a guest and their member in a dedicated channel created from livechat
-                __, guest = channel.sudo()._find_or_create_persona_for_channel(
-                    guest_name=self._get_guest_name(),
+                channel.sudo()._find_or_create_persona_for_channel(
```

Comment:
Since we already called `_get_or_create_guest`, we know we have a valid guest or partner at this point. The only reason we're calling `_find_or_create_persona_for_channel` is because `_get_livechat_channel_vals` doesn’t consistently add the member (only if the user isn’t public).

I think we should clean this up. Either we always add the member in `_get_livechat_channel_vals` and drop this extra logic, or we never do it there and always call `_add_members` instead. That way, we handle member creation in a consistent way.